### PR TITLE
feat: adding vendoring

### PR DESCRIPTION
**Pull request by jeff-roche**

### CODE REVIEW

**Comment by ebblake** [comment]
Can you please amend the commit message to give more rationale about this?  I personally get why you want this: vendoring means the repository is now stand-alone, and any time we upgrade a dependency, it now shows in the repo without CI needing to access the web - but the commit message should state it for the sake of someone less familiar with the why.

Also, if you can include a Signed-off-by tag (with DCO semantics used by the Linux kernel), it will make it easier to merge this without waiting for Red Hat legal to state whether it will be mandatory for patches going forward in this project.

**Comment by ebblake** [comment]
One more thing - this looks like a very automated patch - can you include the command line(s) that you ran to produce this patch, so that it is easy to reproduce?  When a patch this large needs to be backported across downstream branches (not that KubeSAN has any yet, but it may in the future), it is much easier to resolve merge conflicts by re-running the generating commands on the downstream branch.

File: Makefile [hunk 1]
```diff
@@ -100,6 +99,14 @@ lint: golangci-lint ## Run the golangci-lint linter.
...
 	$(GOLANGCI_LINT) run --fix
 
+.PHONY: tidy
+tidy:
+	go mod tidy
+
+.PHONY: vendor
+vendor: tidy
+	go mod vendor
+
```
Comment (ebblake) [inline Makefile:109]: Looks like you patched Makefile (a one-time manual step), and then 'go mod vendor' becomes the command to automate the rest of the patch (a step that can be repeated in the future when we bump to newer dependencies).  Splitting that into two patches may make it easier to distinguish the manual changes from the automated ones.

File: go.mod [hunk 2]
```diff
@@ -6,8 +6,7 @@ toolchain go1.22.2
 
 require (
 	github.com/container-storage-interface/spec v1.9.0
-	github.com/onsi/ginkgo/v2 v2.17.2
-	github.com/onsi/gomega v1.33.1
+	github.com/digitalocean/go-qemu v0.0.0-20230711162256-2e3d0186973e
 	github.com/openshift/custom-resource-status v1.1.2
```
Comment (ebblake) [inline go.mod:10]: I'm trying to figure out why vendoring changed go.mod.  Do we need a separate patch to clean up go.mod before vendoring?

File: Makefile (hunk 1 above)
Comment (jeff-roche) [reply Makefile:109]: I would not separate these, I used `make vendor` to generate this entire PR

File: go.mod (hunk 2 above)
Comment (jeff-roche) [reply go.mod:10]: It's good practice when using go modules to tidy. Tidying will make sure you have all the proper dependencies listed in your go.mod file. You can then run vendor to put all of the dependencies into the vendor folder (automated with `go mod vendor`)

Comment (awels) [reply go.mod:10]: The usual sequence of commands is `go mod tidy`, then `go mod vendor`

Comment (jeff-roche) [reply go.mod:10]: ^ yes

**Comment by jeff-roche** [comment]
Closing in favor of upstream MR https://gitlab.com/kubesan/kubesan/-/merge_requests/109
